Kernel/Problem.cpp

```diff
@@ -303,6 +301,7 @@ void Problem::readDetailsFromProperty() const
   _quantifiesOverPolymorphicVar = _property->quantifiesOverPolymorphicVar();
   _hasBoolVar = _property->hasBoolVar();
   _higherOrder = _property->higherOrder();
+  _hasNonDefaultSorts = _property->hasNonDefaultSorts();
```

Comment:
Depending on the size of diff I'd be somewhat inclined to not lift these up to `Problem` - something else to keep synchronised. What do you think?

Reply:
It felt quite natural for this boolean one. But I already backtracked the lifts of "category" and "usesSorts" which I originally started as well..